contrib/http_util/src/main/java/io/opencensus/contrib/http/AbstractHttpHandler.java

```diff
@@ -66,36 +69,34 @@ private static void putAttributeIfNotEmptyOrNull(Span span, String key, @Nullabl
   }
 
   /**
-   * Instrument an HTTP span after a message is sent.
+   * Instrument an HTTP span after a message is sent. Typically called when last chunk of request or
+   * response is sent.
    *
-   * @param span the span.
-   * @param messageId an id for the message.
-   * @param messageSize the size of the message.
+   * @param context request specific {@link HttpRequestContext}
    * @since 0.18
    */
-  // [TODO:rghetia] add it back after 0.18 is released
-  final void handleMessageSent(Span span, long messageId, long messageSize) {
-    checkNotNull(span, "span");
-    if (span.getOptions().contains(Options.RECORD_EVENTS)) {
+  public final void handleMessageSent(HttpRequestContext context, long bytes) {
+    checkNotNull(context, "context");
+    context.sentMessageSize.addAndGet(bytes);
+    if (context.span.getOptions().contains(Options.RECORD_EVENTS)) {
       // record compressed size
-      recordMessageEvent(span, messageId, Type.SENT, messageSize, 0L);
+      recordMessageEvent(context.span, context.reqId, Type.SENT, bytes, 0L);
     }
   }
 
   /**
-   * Instrument an HTTP span after a message is received.
+   * Instrument an HTTP span after a message is received. Typically called when last chunk of
+   * request or response is received.
    *
-   * @param span the span.
-   * @param messageId an id for the message.
-   * @param messageSize the size of the message.
+   * @param context request specific {@link HttpRequestContext}
    * @since 0.18
    */
-  // [TODO:rghetia] add it back after 0.18 is released
-  final void handleMessageReceived(Span span, long messageId, long messageSize) {
-    checkNotNull(span, "span");
-    if (span.getOptions().contains(Options.RECORD_EVENTS)) {
+  public final void handleMessageReceived(HttpRequestContext context, long bytes) {
+    checkNotNull(context, "context");
+    context.receiveMessageSize.addAndGet(bytes);
+    if (context.span.getOptions().contains(Options.RECORD_EVENTS)) {
       // record compressed size
-      recordMessageEvent(span, messageId, Type.RECEIVED, messageSize, 0L);
+      recordMessageEvent(context.span, context.reqId, Type.RECEIVED, bytes, 0L);
```

Comment:
Should we have to ids one for received and one for sent messages?

Reply:
what is the purpose of ids? Is it more like a sequence number rather than an id? Not sure how it will be used

Reply:
You need to match the message sent with the message received in order to determine network issues or other things.

Reply:
done.